internal/querycoord/cluster.go

```diff
@@ -546,6 +549,80 @@ func (c *queryNodeCluster) getMetrics(ctx context.Context, in *milvuspb.GetMetri
 	return ret
 }
 
+func (c *queryNodeCluster) setNodeState(nodeID int64, node Node, state nodeState) {
+	if state == offline {
```

Comment:
better to extract this logic into a separate function and call it outside the setNodeState function。